<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Expression Weighted Celltype Enrichment • EWCE (Expression Weighted Celltype Enrichment)</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="bootstrap-toc.css">
<script src="bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="pkgdown.css" rel="stylesheet">
<script src="pkgdown.js"></script><meta property="og:title" content="Expression Weighted Celltype Enrichment">
<meta property="og:description" content="Used to determine which cell types are enriched within gene lists. The package
        provides tools for testing enrichments within simple gene lists (such as human disease
        associated genes) and those resulting from differential expression studies. The package does
        not depend upon any particular Single Cell Transcriptome dataset and user defined datasets
        can be loaded in and used in the analyses.">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><!-- Global site tag (gtag.js) - Google Analytics --><script async src="https://www.googletagmanager.com/gtag/js?id=UA-175929032-1"></script><script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-175929032-1');
</script>
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-home">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="index.html">EWCE (Expression Weighted Celltype Enrichment)</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.99.2</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="articles/EWCE.html">Get started</a>
</li>
<li>
  <a href="reference/index.html">Reference</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="contents col-md-9">

<div id="expression-weighted-celltype-enrichment-with-ewce" class="section level1">
<div class="page-header"><h1 class="hasAnchor">
<a href="#expression-weighted-celltype-enrichment-with-ewce" class="anchor"></a>Expression Weighted Celltype Enrichment with <em>EWCE</em>
</h1></div>
<p>Nathan Skene 2020-08-20</p>
<!-- Readme.md is generated from Readme.Rmd. Please edit that file -->
<!-- badges: start -->
<p><a href="https://travis-ci.org/NathanSkene/EWCE"><img src="https://travis-ci.org/NathanSkene/EWCE.svg?branch=master" alt="Build Status"></a> <!-- badges: end --></p>
</div>
<div id="introduction" class="section level1">
<h1 class="hasAnchor">
<a href="#introduction" class="anchor"></a>Introduction</h1>
<p>The <em>EWCE</em> package is designed to facilitate expression weighted celltype enrichment analysis as described in our Frontiers in Neuroscience paper.<sup>1</sup></p>
<p>The package was originally designed to work with the single cell cortical transcriptome data from the Linnarsson lab<sup>2</sup> which is available at <a href="http://linnarssonlab.org/cortex/" class="uri">http://linnarssonlab.org/cortex/</a>. Using this package it is possible to read in any single cell transcriptome data, provided that you have a cell by gene expression matrix (with each cell as a seperate column) and a seperate annotation dataframe, with a row for each cell.</p>
<p>The <em>EWCE</em> process involves testing for whether the genes in a target list have higher levels of expression in a given cell type than can reasonably be expected by chance. The probability distribution for this is estimated by randomly generating gene lists of equal length from a set of background genes.</p>
<p>The <em>EWCE</em> method can be applied to any gene list. In the paper we reported it’s application to genetic and transcriptomic datasets, and in this vignette we detail how this can be done.</p>
<p>Note that throughout this vignette we use the terms ‘cell type’ and ‘sub-cell type’ to refer to two levels of annotation of what a cell type is. This is described in further detail in our paper<sup>1</sup>, but relates to the two levels of annotation provided in the Linnarsson dataset<sup>2</sup>. In this dataset a cell is described as having a cell type (i.e. ‘Interneuron’) and subcell type (i.e. ‘Int11’ a.k.a Interneuron type 11).</p>
</div>
<div id="overview" class="section level1">
<h1 class="hasAnchor">
<a href="#overview" class="anchor"></a>Overview</h1>
<p>The process for using <em>EWCE</em> essentially involves three steps.</p>
<p>First, one needs to load the relevant single cell transcriptome dataset. Single cell transcriptome data is read in from a text file using the <code>read_celltype_data</code>.</p>
<p>The user then obtains a gene set and a suitable background gene set. As the choice of gene sets is up to the user we do not provide functions for doing this. Appropriate choice of background set is discussed in the associated publication.</p>
<p>Bootstrapping is then performed using the <code>bootstrap.enrichment.test</code> function.</p>
</div>
<div id="installing-ewce" class="section level1">
<h1 class="hasAnchor">
<a href="#installing-ewce" class="anchor"></a>Installing EWCE</h1>
<p>The <em>EWCE</em> package is available from github. The old version available from the Bioconductor archives is depreciated and should not be used. To be able to install the package one needs first to install R then run the following lines of code:</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode R"><code class="sourceCode r"><a class="sourceLine" id="cb1-1" data-line-number="1"><span class="cf">if</span> (<span class="op">!</span><span class="kw">require</span>(<span class="st">"devtools"</span>)) {</a>
<a class="sourceLine" id="cb1-2" data-line-number="2">  <span class="kw">install.packages</span>(<span class="st">"devtools"</span>)</a>
<a class="sourceLine" id="cb1-3" data-line-number="3">}</a>
<a class="sourceLine" id="cb1-4" data-line-number="4">devtools<span class="op">::</span><span class="kw">install_github</span>(<span class="st">"nathanskene/ewce"</span>)</a></code></pre></div>
<p>You can then load the package:</p>
<div class="sourceCode" id="cb2"><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">EWCE</span>)</pre></div>
</div>
<div id="using-with-docker" class="section level1">
<h1 class="hasAnchor">
<a href="#using-with-docker" class="anchor"></a>Using with docker</h1>
<p>Images with the latest version of EWCE are regularly pushed to Dockerhub. If you already have docker installed you can load up a working copy using the following commands. Note, that you will need to replace the initial directory path with a location on your computer that you wish to be able to access from within the docker image.</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode R"><code class="sourceCode r"><a class="sourceLine" id="cb3-1" data-line-number="1">docker pull nathanskene<span class="op">/</span>ewce</a>
<a class="sourceLine" id="cb3-2" data-line-number="2">docker run <span class="op">--</span>name=ewce <span class="op">-</span>e PASSWORD=ewcedocker <span class="op">-</span>p <span class="dv">8790</span><span class="op">:</span><span class="dv">8790</span> <span class="op">-</span>d <span class="op">-</span>v <span class="op">/</span>User<span class="op">/</span><span class="er">$</span>USER<span class="op">:</span><span class="er">/</span>var<span class="op">/</span>ewce nathanskene<span class="op">/</span>ewce<span class="op">:</span>latest</a>
<a class="sourceLine" id="cb3-3" data-line-number="3">docker exec <span class="op">-</span>ti ewce R</a></code></pre></div>
</div>
<div id="getting-started" class="section level1">
<h1 class="hasAnchor">
<a href="#getting-started" class="anchor"></a>Getting started</h1>
<p>See the <a href="https://nathanskene.github.io/EWCE/articles/EWCE.html">vignette website</a> for up-to-date instructions on usage.</p>
<p>If you have any problems please do file an issue here on github.</p>
</div>
<div id="citation" class="section level1">
<h1 class="hasAnchor">
<a href="#citation" class="anchor"></a>Citation</h1>
<p>If you use the EWCE package as well then please cite</p>
<p><a href="https://www.frontiersin.org/articles/10.3389/fnins.2016.00016/full">Skene, et al. Identification of Vulnerable Cell Types in Major Brain Disorders Using Single Cell Transcriptomes and Expression Weighted Cell Type Enrichment. Front. Neurosci, 2016.</a></p>
<p>If you use the cortex/hippocampus single cell data associated with this package then please cite the following papers:</p>
<p><a href="http://www.sciencemag.org/content/early/2015/02/18/science.aaa1934.abstract">Zeisel, et al. Cell types in the mouse cortex and hippocampus revealed by single-cell RNA-seq. Science, 2015.</a></p>
</div>
<div id="references" class="section level1">
<h1 class="hasAnchor">
<a href="#references" class="anchor"></a>References</h1>
<div id="refs" class="references">
<div id="ref-skene_2016">
<p>1. Skene, N. &amp; Grant, S. Identification of vulnerable cell types in major brain disorders using single cell transcriptomes and expression weighted cell type enrichment. <em>Frontiers in Neuroscience</em> (2016). <a href="doi:%5B10.3389/fnins.2016.00016%5D(https://doi.org/10.3389/fnins.2016.00016)" class="uri">doi:[10.3389/fnins.2016.00016](https://doi.org/10.3389/fnins.2016.00016)</a></p>
</div>
<div id="ref-zeisel2015cell">
<p>2. Zeisel, A. <em>et al.</em> Cell types in the mouse cortex and hippocampus revealed by single-cell rna-seq. <em>Science</em> <strong>347,</strong> 1138–1142 (2015).</p>
</div>
</div>
</div>

  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <div class="license">
<h2>License</h2>
<ul class="list-unstyled">
<li><a href="https://www.r-project.org/Licenses/Artistic-2.0">Artistic-2.0</a></li>
</ul>
</div>
<div class="developers">
<h2>Developers</h2>
<ul class="list-unstyled">
<li>Nathan Skene <br><small class="roles"> Maintainer </small>  </li>
</ul>
</div>

  </div>
</div>


      <footer><div class="copyright">
  <p>Developed by Nathan Skene.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.5.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
