<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Bootstrap celltype enrichment test — bootstrap.enrichment.test • EWCE (Expression Weighted Celltype Enrichment)</title>


<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->
<link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous" />


<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="Bootstrap celltype enrichment test — bootstrap.enrichment.test" />
<meta property="og:description" content="bootstrap.enrichment.test takes a genelist and a single cell type transcriptome dataset
and determines the probability of enrichment and fold changes for each cell type." />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->


<!-- Global site tag (gtag.js) - Google Analytics -->
<script async src="https://www.googletagmanager.com/gtag/js?id=UA-175929032-1"></script>
<script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-175929032-1');
</script>


  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">EWCE (Expression Weighted Celltype Enrichment)</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.99.2</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/EWCE.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Bootstrap celltype enrichment test</h1>
    
    <div class="hidden name"><code>bootstrap.enrichment.test.Rd</code></div>
    </div>

    <div class="ref-description">
    <p><code>bootstrap.enrichment.test</code> takes a genelist and a single cell type transcriptome dataset
and determines the probability of enrichment and fold changes for each cell type.</p>
    </div>

    <pre class="usage"><span class='fu'>bootstrap.enrichment.test</span>(
  sct_data = <span class='fl'>NA</span>,
  hits = <span class='fl'>NA</span>,
  bg = <span class='fl'>NA</span>,
  genelistSpecies = <span class='st'>"mouse"</span>,
  sctSpecies = <span class='st'>"mouse"</span>,
  reps = <span class='fl'>100</span>,
  annotLevel = <span class='fl'>1</span>,
  geneSizeControl = <span class='fl'>FALSE</span>,
  controlledCT = <span class='kw'>NULL</span>
)</pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>sct_data</th>
      <td><p>List generated using <code><a href='generate.celltype.data.html'>generate.celltype.data</a></code></p></td>
    </tr>
    <tr>
      <th>hits</th>
      <td><p>Array of MGI gene symbols containing the target gene list. Must be HGNC symbols if geneSizeControl=TRUE</p></td>
    </tr>
    <tr>
      <th>bg</th>
      <td><p>Array of MGI gene symbols containing the background gene list. Must be HGNC symbols if geneSizeControl=TRUE</p></td>
    </tr>
    <tr>
      <th>genelistSpecies</th>
      <td><p>Either 'mouse' or 'human' depending on whether MGI or HGNC symbols are used for gene lists</p></td>
    </tr>
    <tr>
      <th>sctSpecies</th>
      <td><p>Either 'mouse' or 'human' depending on whether MGI or HGNC symbols are used for the single cell dataset</p></td>
    </tr>
    <tr>
      <th>reps</th>
      <td><p>Number of random gene lists to generate (default=100 but should be over 10000 for publication quality results)</p></td>
    </tr>
    <tr>
      <th>annotLevel</th>
      <td><p>an integer indicating which level of the annotation to analyse. Default = 1.</p></td>
    </tr>
    <tr>
      <th>geneSizeControl</th>
      <td><p>a logical indicating whether you want to control for GC content and transcript length. Recommended if the gene list originates from genetic studies. Default is FALSE. If set to TRUE then human gene lists should be used rather than mouse.</p></td>
    </tr>
    <tr>
      <th>controlledCT</th>
      <td><p>(optional) If not NULL, and instead is the name of a cell type, then the bootstrapping controls for expression within that celltype</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>A list containing three data frames:</p><ul>
<li><p><code>results</code>: dataframe in which each row gives the statistics (p-value, fold change and number of standard deviations from the mean) associated with the enrichment of the stated cell type in the gene list</p></li>
<li><p><code>hit.cells</code>: vector containing the summed proportion of expression in each cell type for the target list</p></li>
<li><p><code>bootstrap_data</code>: matrix in which each row represents the summed proportion of expression in each cell type for one of the random lists</p></li>
</ul>


    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'><span class='co'>if</span> (<span class='fl'>FALSE</span>) {
<span class='co'># Load the single cell data</span>
<span class='fu'><a href='https://rdrr.io/r/utils/data.html'>data</a></span>(<span class='st'>"ctd"</span>)

<span class='co'># Set the parameters for the analysis</span>
<span class='kw'>reps</span><span class='op'>=</span><span class='fl'>100</span> <span class='co'># &lt;- Use 100 bootstrap lists so it runs quickly, for publishable analysis use &gt;10000</span>

<span class='co'># Load the gene list and get human orthologs</span>
<span class='fu'><a href='https://rdrr.io/r/utils/data.html'>data</a></span>(<span class='st'>"example_genelist"</span>)
<span class='fu'><a href='https://rdrr.io/r/utils/data.html'>data</a></span>(<span class='st'>"mouse_to_human_homologs"</span>)
<span class='kw'>m2h</span> <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/unique.html'>unique</a></span>(<span class='kw'>mouse_to_human_homologs</span>[,<span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span>(<span class='st'>"HGNC.symbol"</span>,<span class='st'>"MGI.symbol"</span>)])
<span class='kw'>mouse.hits</span> <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/unique.html'>unique</a></span>(<span class='kw'>m2h</span>[<span class='kw'>m2h</span><span class='op'>$</span><span class='kw'>HGNC.symbol</span> <span class='op'>%in%</span> <span class='kw'>example_genelist</span>,<span class='st'>"MGI.symbol"</span>])
<span class='kw'>human.hits</span> <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/unique.html'>unique</a></span>(<span class='kw'>m2h</span>[<span class='kw'>m2h</span><span class='op'>$</span><span class='kw'>HGNC.symbol</span> <span class='op'>%in%</span> <span class='kw'>example_genelist</span>,<span class='st'>"HGNC.symbol"</span>])
<span class='kw'>human.bg</span> <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/unique.html'>unique</a></span>(<span class='kw'>m2h</span><span class='op'>$</span><span class='kw'>HGNC.symbol</span>)
<span class='kw'>mouse.bg</span>  <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/unique.html'>unique</a></span>(<span class='kw'>m2h</span><span class='op'>$</span><span class='kw'>MGI.symbol</span>)

<span class='co'># Bootstrap significance testing, without controlling for transcript length and GC content</span>
<span class='kw'>full_results</span> <span class='op'>=</span> <span class='fu'>bootstrap.enrichment.test</span>(sct_data=<span class='kw'>ctd</span>,hits=<span class='kw'>mouse.hits</span>,
  bg=<span class='kw'>mouse.bg</span>,reps=<span class='kw'>reps</span>,annotLevel=<span class='fl'>2</span>,sctSpecies=<span class='st'>"mouse"</span>,genelistSpecies=<span class='st'>"mouse"</span>)

<span class='co'># Bootstrap significance testing controlling for transcript length and GC content</span>
<span class='kw'>full_results</span> <span class='op'>=</span> <span class='fu'>bootstrap.enrichment.test</span>(sct_data=<span class='kw'>ctd</span>,hits=<span class='kw'>human.hits</span>,
  bg=<span class='kw'>human.bg</span>,reps=<span class='kw'>reps</span>,annotLevel=<span class='fl'>2</span>,geneSizeControl=<span class='fl'>TRUE</span>,
  sctSpecies=<span class='st'>"mouse"</span>,genelistSpecies=<span class='st'>"human"</span>)
}
</div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by Nathan Skene.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.0.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


