<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Generate bootstrap plots — generate.bootstrap.plots.for.transcriptome • EWCE</title>


<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous" />

<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="Generate bootstrap plots — generate.bootstrap.plots.for.transcriptome" />
<meta property="og:description" content="generate.bootstrap.plots.for.transcriptome takes a genelist and a single cell type transcriptome dataset
and generates plots which show how the expression of the genes in the list compares to those
in randomly generated gene lists" />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">EWCE</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.99.2</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/EWCE.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Generate bootstrap plots</h1>
    
    <div class="hidden name"><code>generate.bootstrap.plots.for.transcriptome.Rd</code></div>
    </div>

    <div class="ref-description">
    <p><code>generate.bootstrap.plots.for.transcriptome</code> takes a genelist and a single cell type transcriptome dataset
and generates plots which show how the expression of the genes in the list compares to those
in randomly generated gene lists</p>
    </div>

    <pre class="usage"><span class='fu'>generate.bootstrap.plots.for.transcriptome</span>(
  <span class='no'>sct_data</span>,
  <span class='no'>tt</span>,
  <span class='kw'>thresh</span> <span class='kw'>=</span> <span class='fl'>250</span>,
  <span class='kw'>annotLevel</span> <span class='kw'>=</span> <span class='fl'>1</span>,
  <span class='no'>reps</span>,
  <span class='kw'>full_results</span> <span class='kw'>=</span> <span class='fl'>NA</span>,
  <span class='kw'>listFileName</span> <span class='kw'>=</span> <span class='st'>""</span>,
  <span class='kw'>showGNameThresh</span> <span class='kw'>=</span> <span class='fl'>25</span>,
  <span class='kw'>ttSpecies</span> <span class='kw'>=</span> <span class='st'>"mouse"</span>,
  <span class='kw'>sctSpecies</span> <span class='kw'>=</span> <span class='st'>"mouse"</span>,
  <span class='kw'>sortBy</span> <span class='kw'>=</span> <span class='st'>"t"</span>,
  <span class='kw'>onlySignif</span> <span class='kw'>=</span> <span class='fl'>TRUE</span>
)</pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>sct_data</th>
      <td><p>List generated using <code><a href='generate.celltype.data.html'>generate.celltype.data</a></code></p></td>
    </tr>
    <tr>
      <th>tt</th>
      <td><p>Differential expression table. Can be output of limma::topTable function. Minimum requirement is that one column stores a metric of increased/decreased expression (i.e. log fold change, t-statistic for differential expression etc) and another contains either HGNC or MGI symbols.</p></td>
    </tr>
    <tr>
      <th>thresh</th>
      <td><p>The number of up- and down- regulated genes to be included in each analysis. Dafault=250</p></td>
    </tr>
    <tr>
      <th>annotLevel</th>
      <td><p>an integer indicating which level of the annotation to analyse. Default = 1.</p></td>
    </tr>
    <tr>
      <th>reps</th>
      <td><p>Number of random gene lists to generate (default=100 but should be over 10000 for publication quality results)</p></td>
    </tr>
    <tr>
      <th>full_results</th>
      <td><p>The full output of <code><a href='bootstrap.enrichment.test.html'>bootstrap.enrichment.test</a></code> for the same genelist</p></td>
    </tr>
    <tr>
      <th>listFileName</th>
      <td><p>String used as the root for files saved using this function</p></td>
    </tr>
    <tr>
      <th>showGNameThresh</th>
      <td><p>Integer. If a gene has over X percent of it's expression proportion in a celltype, then list the gene name</p></td>
    </tr>
    <tr>
      <th>ttSpecies</th>
      <td><p>Either 'mouse' or 'human' depending on which species the differential expression table was generated from</p></td>
    </tr>
    <tr>
      <th>sctSpecies</th>
      <td><p>Either 'mouse' or 'human' depending on which species the single cell data was generated from</p></td>
    </tr>
    <tr>
      <th>sortBy</th>
      <td><p>Column name of metric in tt which should be used to sort up- from down- regulated genes. Default="t"</p></td>
    </tr>
    <tr>
      <th>onlySignif</th>
      <td><p>Should plots only be generated for cells which have significant changes?</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>Saves a set of pdf files containing graphs. These will be saved with the filename adjusted using the
value of listFileName. The files are saved into the 'BootstrapPlot' folder. The files start with one of the following:</p><ul>
<li><p><code>qqplot_noText</code>: sorts the gene list according to how enriched it is in the relevant celltype. Plots the value in the target list against the mean value in the bootstrapped lists.</p></li>
<li><p><code>qqplot_wtGSym</code>: as above but labels the gene symbols for the highest expressed genes.</p></li>
<li><p><code>bootDists</code>: rather than just showing the mean of the bootstrapped lists, a boxplot shows the distribution of values</p></li>
<li><p><code>bootDists_LOG</code>: shows the bootstrapped distributions with the y-axis shown on a log scale</p></li>
</ul>


    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'><span class='kw'>if</span> (<span class='fl'>FALSE</span>) {
<span class='co'># Load the single cell data</span>
<span class='fu'><a href='https://rdrr.io/r/utils/data.html'>data</a></span>(<span class='st'>"ctd"</span>)

<span class='co'># Set the parameters for the analysis</span>
<span class='no'>reps</span><span class='kw'>=</span><span class='fl'>100</span> <span class='co'># &lt;- Use 100 bootstrap lists so it runs quickly, for publishable analysis use &gt;10000</span>
<span class='no'>annotLevel</span><span class='kw'>=</span><span class='fl'>1</span> <span class='co'># &lt;- Use cell level annotations (i.e. Interneurons)</span>

<span class='co'># Load the top table</span>
<span class='fu'><a href='https://rdrr.io/r/utils/data.html'>data</a></span>(<span class='no'>tt_alzh</span>)

<span class='no'>tt_results</span> <span class='kw'>=</span> <span class='fu'><a href='ewce_expression_data.html'>ewce_expression_data</a></span>(<span class='kw'>sct_data</span><span class='kw'>=</span><span class='no'>ctd</span>,<span class='kw'>tt</span><span class='kw'>=</span><span class='no'>tt_alzh</span>,<span class='kw'>annotLevel</span><span class='kw'>=</span><span class='fl'>1</span>,
    <span class='kw'>ttSpecies</span><span class='kw'>=</span><span class='st'>"human"</span>,<span class='kw'>sctSpecies</span><span class='kw'>=</span><span class='st'>"mouse"</span>)

<span class='co'># Bootstrap significance testing, without controlling for transcript length and GC content</span>
<span class='no'>full_results</span> <span class='kw'>=</span> <span class='fu'>generate.bootstrap.plots.for.transcriptome</span>(<span class='kw'>sct_data</span><span class='kw'>=</span><span class='no'>ctd</span>,<span class='kw'>tt</span><span class='kw'>=</span><span class='no'>tt_alzh</span>,<span class='kw'>annotLevel</span><span class='kw'>=</span><span class='fl'>1</span>,
     <span class='kw'>full_results</span><span class='kw'>=</span><span class='no'>tt_results</span>,<span class='kw'>listFileName</span><span class='kw'>=</span><span class='st'>"examples"</span>,<span class='kw'>reps</span><span class='kw'>=</span><span class='no'>reps</span>,<span class='kw'>ttSpecies</span><span class='kw'>=</span><span class='st'>"human"</span>,<span class='kw'>sctSpecies</span><span class='kw'>=</span><span class='st'>"mouse"</span>)
}</div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by Nathan Skene.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.5.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


