<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Function reference â€¢ EWCE (Expression Weighted Celltype Enrichment)</title>


<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->
<link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous" />


<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="Function reference" />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->


<!-- Global site tag (gtag.js) - Google Analytics -->
<script async src="https://www.googletagmanager.com/gtag/js?id=UA-175929032-1"></script>
<script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-175929032-1');
</script>


  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-index">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">EWCE (Expression Weighted Celltype Enrichment)</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.99.2</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/EWCE.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="contents col-md-9">
    <div class="page-header">
      <h1>Reference</h1>
    </div>

    <table class="ref-index">

    <colgroup>
      
      <col class="alias" />
      <col class="title" />
    </colgroup>

    <tbody>
      <tr>
        <th colspan="2">
          <h2 id="section-all-functions" class="hasAnchor"><a href="#section-all-functions" class="anchor"></a>All functions</h2>
          <p class="section-desc"></p>
        </th>
      </tr>
      
      
    </tbody><tbody>
      
      
      <tr>
        
        <td>
          <p><code><a href="add.res.to.merging.list.html">add.res.to.merging.list()</a></code> </p>
        </td>
        <td><p>Add to results to merging list</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="all_hgnc.html">all_hgnc</a></code> </p>
        </td>
        <td><p>All HGNC gene symbols</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="all_hgnc_wtEnsembl.html">all_hgnc_wtEnsembl</a></code> </p>
        </td>
        <td><p>All HGNC gene symbols with ENSEMBL gene IDs</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="all_mgi.html">all_mgi</a></code> </p>
        </td>
        <td><p>All MGI gene symbols</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="all_mgi_wtEnsembl.html">all_mgi_wtEnsembl</a></code> </p>
        </td>
        <td><p>All MGI gene symbols with ENSEMBL gene IDs</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="alzh_gwas_top100.html">alzh_gwas_top100</a></code> </p>
        </td>
        <td><p>Alzheimers disease top100 GWAS genes generated by MAGMA using the iGAP summary statistics</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="bin.columns.into.quantiles.html">bin.columns.into.quantiles()</a></code> </p>
        </td>
        <td><p>bin.columns.into.quantiles</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="bin.specificity.into.quantiles.html">bin.specificity.into.quantiles()</a></code> </p>
        </td>
        <td><p>bin.specificity.into.quantiles</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="bootstrap.enrichment.test.html">bootstrap.enrichment.test()</a></code> </p>
        </td>
        <td><p>Bootstrap celltype enrichment test</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="check.ewce.genelist.inputs.html">check.ewce.genelist.inputs()</a></code> </p>
        </td>
        <td><p>check.ewce.genelist.inputs</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="controlled_geneset_enrichment.html">controlled_geneset_enrichment()</a></code> </p>
        </td>
        <td><p>Celltype controlled geneset enrichment</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="convert_new_ewce_to_old.html">convert_new_ewce_to_old()</a></code> </p>
        </td>
        <td><p>convert_new_ewce_to_old</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="convert_old_ewce_to_new.html">convert_old_ewce_to_new()</a></code> </p>
        </td>
        <td><p>convert_old_ewce_to_new</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="cortex_mrna.html">cortex_mrna</a></code> </p>
        </td>
        <td><p>Karolinska Cortex/Hippocamus dataset subsample</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="ctd.html">ctd</a></code> </p>
        </td>
        <td><p>The genes from Karolinska cortex/hippocampus and hypothalamus single cell transcriptome datasets</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="drop.uninformative.genes.html">drop.uninformative.genes()</a></code> </p>
        </td>
        <td><p>drop.uninformative.genes</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="ensembl_transcript_lengths_GCcontent.html">ensembl_transcript_lengths_GCcontent</a></code> </p>
        </td>
        <td><p>Human Ensembl Transcript Lengths &amp; GC content</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="ewce.plot.html">ewce.plot()</a></code> </p>
        </td>
        <td><p>Plot EWCE results</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="ewce_expression_data.html">ewce_expression_data()</a></code> </p>
        </td>
        <td><p>Bootstrap celltype enrichment test for transcriptome data</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="example_genelist.html">example_genelist</a></code> </p>
        </td>
        <td><p>Example Gene list</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="filter.genes.without.1to1.homolog.html">filter.genes.without.1to1.homolog()</a></code> </p>
        </td>
        <td><p>filter.genes.without.1to1.homolog</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="fix.bad.hgnc.symbols.html">fix.bad.hgnc.symbols()</a></code> </p>
        </td>
        <td><p>fix.bad.hgnc.symbols
- Given an expression matrix, wherein the rows are supposed to be HGNC symbols, find those symbols which are not official HGNC symbols, then
correct them if possible. Return the expression matrix with corrected symbols.</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="fix.bad.mgi.symbols.html">fix.bad.mgi.symbols()</a></code> </p>
        </td>
        <td><p>fix.bad.mgi.symbols
- Given an expression matrix, wherein the rows are supposed to be MGI symbols, find those symbols which are not official MGI symbols, then
check in the MGI synonm database for whether they match to a proper MGI symbol. Where a symbol is found to be an aliases for a gene that is already
in the dataset, the combined reads are summed together.</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="generate.bootstrap.plots.html">generate.bootstrap.plots()</a></code> </p>
        </td>
        <td><p>Generate bootstrap plots</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="generate.bootstrap.plots.for.transcriptome.html">generate.bootstrap.plots.for.transcriptome()</a></code> </p>
        </td>
        <td><p>Generate bootstrap plots</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="generate.celltype.data.html">generate.celltype.data()</a></code> </p>
        </td>
        <td><p>generate.celltype.data</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="generate_controlled_bootstrap_geneset.html">generate_controlled_bootstrap_geneset()</a></code> </p>
        </td>
        <td><p>generate_controlled_bootstrap_geneset</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="get.celltype.table.html">get.celltype.table()</a></code> </p>
        </td>
        <td><p>get.celltype.table</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="get.length.of.longest.string.html">get.length.of.longest.string()</a></code> </p>
        </td>
        <td><p>get.length.of.longest.string</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="get_summed_proportions.html">get_summed_proportions()</a></code> </p>
        </td>
        <td><p>Get summed proportions</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="hpsd_genes.html">hpsd_genes</a></code> </p>
        </td>
        <td><p>Human post-synaptic density</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="id_genes.html">id_genes</a></code> </p>
        </td>
        <td><p>Intellectual disability genes</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="load.linnarsson.sct.data.html">load.linnarsson.sct.data()</a></code> </p>
        </td>
        <td><p>load.linnarsson.sct.data</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="merge_two_expfiles.html">merge_two_expfiles()</a></code> </p>
        </td>
        <td><p>merge_two_expfiles</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="merged_ewce.html">merged_ewce()</a></code> </p>
        </td>
        <td><p>Multiple EWCE results from multiple studies</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="mgi_synonym_data.html">mgi_synonym_data</a></code> </p>
        </td>
        <td><p>MGI synonym data</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="mouse_to_human_homologs.html">mouse_to_human_homologs</a></code> </p>
        </td>
        <td><p>Table of Human--&gt;Mouse orthologs for all human genes</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="prep.dendro.html">prep.dendro()</a></code> </p>
        </td>
        <td><p>prep.dendro</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="prepare.genesize.control.network.html">prepare.genesize.control.network()</a></code> </p>
        </td>
        <td><p>Prepare genesize control network</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="rbfox_genes.html">rbfox_genes</a></code> </p>
        </td>
        <td><p>Rbfox binding genes</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="schiz_genes.html">schiz_genes</a></code> </p>
        </td>
        <td><p>Schizophrenia susceptibility genes from CLOZUK</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="tt_alzh.html">tt_alzh</a></code> </p>
        </td>
        <td><p>Example table of differential expression (Alzheimer's BA46)</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="tt_alzh_BA36.html">tt_alzh_BA36</a></code> </p>
        </td>
        <td><p>Example table of differential expression (Alzheimer's BA36)</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="tt_alzh_BA44.html">tt_alzh_BA44</a></code> </p>
        </td>
        <td><p>Example table of differential expression (Alzheimer's BA44)</p></td>
      </tr>
    </tbody>
    </table>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by Nathan Skene.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.5.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


